### PR TITLE
Initial auto PR labeler config

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Along the same lines as #7091, however using a cron based action to work round permission issues.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+core:
+  - quantum/**/*
+  - tmk_core/**/*
+  - drivers/**/*
+  - tests/**/*
+  - util/**/*
+keymap:
+  - users/**/*
+  - layouts/**/*
+  - keyboards/**/keymap/**/*
```
Comment (noroadsleft) [inline .github/labeler.yml:10]:
```suggestion
  - keyboards/**/keymaps/**/*
```

**Comment by tzarc** [comment]
Are we able to retarget to `develop` if there are core changes?

File: .github/labeler.yml (hunk 1 above)
Comment (drashna) [reply .github/labeler.yml:10]: Is it possible to exclude `default` and `via` here?

Comment (zvecr) [reply .github/labeler.yml]: Maybe in the future but not right now as it has the same limitations as the labeler actions

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+core:
+  - quantum/**/*
+  - tmk_core/**/*
+  - drivers/**/*
+  - tests/**/*
+  - util/**/*
```
Comment (tzarc) [inline .github/labeler.yml:6]: In prep for `develop`'s merge, should we get this in?
```suggestion
  - util/**/*
  - platforms/**/*
```

**Pull request merged**